src/multidim/intervalbox.jl

```diff
@@ -138,6 +138,26 @@ as a vector.
     end
 end
 
+@generated function mince(x::IntervalBox{N,T}, tup::NTuple{N,Int}) where {N,T}
+    quote
+        n = maximum(tup)
+        nodes_matrix = Array{Interval{T},2}(undef, n, N)
+        fill!(nodes_matrix, emptyinterval(T))
+        for i in 1:N
+            nodes_matrix[1:tup[i],i] .= mince(x[i], tup[i])
+        end
+
+        nodes = IntervalBox{$N,T}[]
+        Base.Cartesian.@nloops $N i _->(1:n) begin
```

Comment:
Does this part requires carefull review? I am not familiar with the macros used, but I could delve into it if needed.

Reply:
Thanks for your review @Kolaru and also for offering getting into the details of this part of the code. 

Here, I simply used the original method, and adapt it to obtain the desired behavior; maybe there are better ways, but for the time being, this is good enough. The original code was inspired from [this section of the manual](https://docs.julialang.org/en/v1/devdocs/cartesian/#Base.Cartesian), which admittedly I have mostly forgotten.

Reply:
That sounded like a fun multidimensional challenge so I spent some time on it and I believe the following is equivalent:
```julia
function mince(x::IntervalBox{N,T}, ncuts::NTuple{N,Int}) where {N,T}
    minced_intervals = [mince(x[i], ncuts[i]) for i in 1:N]
    minced_boxes = Vector{IntervalBox{N,T}}(undef, prod(ncuts))

    for (k, cut_indices) in enumerate(CartesianIndices(ncuts))
        ibox = IntervalBox([minced_intervals[i][cut_indices[i]] for i in 1:N])
        minced_boxes[k] = ibox
    end
    return minced_boxes
end
```

Reply:
Thanks @Kolaru for the suggestion! I am also changing the other method (avoiding then the `@generated`). I would like to [include you as coauthor](https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#required-co-author-information) of this commit; is it ok with you? (I already have the commit ready, but prefer to wait for your reply before pushing it.)

Reply:
Sure let me know if it needs any input from my side.

Reply:
I just pushed it, so you can have a look